### ISSUE TITLE
record update fails with message refused

### ISSUE THREAD

**Issue by dckc** [monte-language/typhon#82]
I ran into this because `lib/http/server` stopped working for my little 15 line http server.

This crashes:

```monte
import "lib/record" =~ [=> makeRecord :DeepFrozen]
exports(main)

def [Headers :DeepFrozen,
     makeHeaders :DeepFrozen] := makeRecord("Headers",
     ["contentLength" => NullOk[Int],
      "contentType" => NullOk[Pair[Str, Str]],
      "spareHeaders" => Map[Str, Str]])

def main(argv) as DeepFrozen:
    def h1 := makeHeaders(null, null, [].asMap())
    traceln(h1)
    def h2 := h1.withSpareHeaders(["Host" => "localhost:5000"])
    traceln(h2)
```

version info: `/nix/store/nh6902ahpzsh2fa352agnn1gp6zj360q-typhon-vm/mt-typhon`

**Comment by dckc** [comment]
That code works in the following versions, which I installed March 15.

0kxzsybaz89d435xf7fjxgqj2sxhxjl5-typhon-vm
42xlh2n91vkckh7cqmmbfw6gr6c1mgcv-typhon

**Comment by MostAwesomeDude** [comment]
I caused the bug. I fixed it in fdb2b52.
